### PR TITLE
Allow disabling buildkit for engines lacking support.

### PR DESCRIPTION
**Pull request by Alexhuszagh**

### CODE REVIEW

File: .changes/1036.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Alexhuszagh) [inline .changes/1036.json:1]: this needs to change to `1032.json`.

**Comment by Alexhuszagh** [comment]
This could also likely be fixed by either using an environment variable (to manually disable) or by checking if `$engine buildx --help` fails, since we've already checked `$engine --version` works at this point. This could use a slightly more comprehensive solution.

**Review by Emilgardis — APPROVED** [review]
this looks good to me for now

adding more comprehensive checks for automatically doing this could be done in the client/server checks for platform I think, only issue is that the fields are not stable between implementations

**Comment by Alexhuszagh** [comment]
bors r=Emilgardis

**Comment by Emilgardis** [comment]
bors retry

possibly spurious error on winehq

File: src/docker/engine.rs [hunk 2]
```diff
@@ -108,6 +108,11 @@ impl Engine {
             .map(|s| bool_from_envvar(&s))
             .unwrap_or_default()
     }
+
+    #[must_use]
+    pub fn has_buildkit() -> bool {
+        !bool_from_envvar("CROSS_CONTAINER_ENGINE_NO_BUILDKIT")
```
Comment (Emilgardis) [inline src/docker/engine.rs:114]: aha... the classic

this functions takes the value of the envvar, not the key...

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: src/docker/engine.rs (hunk 2 above)
Comment (Emilgardis) [reply src/docker/engine.rs:114]: we should look into making this harder to missuse

Comment (Alexhuszagh) [reply src/docker/engine.rs:114]: I thought for consistency we should keep it as `0`, `1`, `true`, `false`, etc.

**Comment by Alexhuszagh** [comment]
> bors retry
> 
> possible spurious error on winehq

It looks like a signing key changed, I'll try to fix this and submit a PR:
https://github.com/cross-rs/cross/actions/runs/3145747924/jobs/5113458177#step:10:1640

**Comment by Emilgardis** [comment]
> It looks like a signing key changed, I'll try to fix this and submit a PR

sounds good!

The zig error is due to not using `buildx` anymore (because `Engine::has_buildkit` always returns false), and thus `TARGETPLATFORM` is not set correctly

https://github.com/cross-rs/cross/actions/runs/3149116522/jobs/5120408064#step:10:1398

**Comment by Alexhuszagh** [comment]
> > It looks like a signing key changed, I'll try to fix this and submit a PR
> 
> sounds good!
> 
> The zig error is due to not using `buildx` anymore (because `Engine::has_buildkit` always returns false), and thus `TARGETPLATFORM` is not set correctly
> 
> https://github.com/cross-rs/cross/actions/runs/3149116522/jobs/5120408064#step:10:1398

Ah ok, so I need to fix that logic then.

File: src/docker/engine.rs (hunk 2 above)
Comment (Emilgardis) [reply src/docker/engine.rs:114]: happened before here: https://github.com/cross-rs/cross/pull/997

Comment (Alexhuszagh) [reply src/docker/engine.rs:114]: Ahhhhh. That is confusing, and I was the one who added that function.

Comment (Alexhuszagh) [reply src/docker/engine.rs]: I've fixed the invocation and also added documentation for `bool_from_envvar`:

```rust
// this takes the value of the environment variable,
// so you should call `bool_from_envvar(env::var("FOO"))`
pub fn bool_from_envvar(envvar: &str) -> bool {
    if let Ok(value) = bool::from_str(envvar) {
        value
    } else if let Ok(value) = i32::from_str(envvar) {
        value != 0
    } else {
        !envvar.is_empty()
    }
}
```

```rust
    #[must_use]
    pub fn has_buildkit() -> bool {
        !env::var("CROSS_CONTAINER_ENGINE_NO_BUILDKIT")
            .map(|x| bool_from_envvar(&x))
            .unwrap_or_default()
    }
```

File: src/config.rs [hunk 3]
```diff
@@ -186,6 +186,8 @@ fn split_to_cloned_by_ws(string: &str) -> Vec<String> {
     string.split_whitespace().map(String::from).collect()
 }
 
+// this takes the value of the environment variable,
+// so you should call `bool_from_envvar(env::var("FOO"))`
```
Comment (Emilgardis) [inline src/config.rs:190]: make this a doc-comment and this should be good to go!

**Review by Emilgardis — APPROVED** [review]

**Comment by Alexhuszagh** [comment]
bors r=Emilgardis

**Pull request merged**